### DISCUSSION TITLE
Unroll for all comparisons

### DISCUSSION THREAD

**Discussion by marek-kuticka** [spockframework/spock#1609] [General]
Good afternoon.
First of all, congrats to all work done for the Spock framework.

My question/idea. similar, as @Unroll annotation enables to trace all iterations of a loop of data in where block, would it be possible to add similar feature into the comparisons as well? Not sure, how much this is connected to Plugin in IntelliJ. there is probably some interface between Spock and Test result view in IntelliJ, what I want to achieve is to be able to specify in the spec, which comparison/test I want to report in detail, which not.

```groovy
def sample() {
    when:
        def a = 1
        def b = 2
    then:
        assert a==1 
        assert b == 2, 'Checking result of variable B' 
}
```

in the result, I want to see:
```plain
'sample' : PASSED
  'Checking result of variable B': PASSED
  (other tests omitted in display) PASSED
```

Unroll annotation "unfortunately" works only for where clauses, and, if possible, I'd like to enhance this a bit.

@pniederw and others, what would you think about this? Danke sehr im Voraus

Kind regards aus Prag

Marek

**Comment by Vampire** [comment]
You should not ping random people, especially people that are not working on the project since years. That's pretty rude.

Do you really need those reported as individual tests? Or do you just want both conditions checked, even if the first failed already? Because for that there is `verifyAll { ... }`.

**Reply by marek-kuticka** [reply]
Hi. I was not aware he is not anymore in project. sorry for this.

to your question, yes I need to have all of them reported

**Reply by leonard84** [reply]
Could you elaborate more, why `verifyAll` is insufficient?
